middle_end/semantics_of_primitives.ml

```diff
@@ -141,6 +141,9 @@ let for_primitive (prim : Clambda_primitives.primitive) =
   | Pdls_get ->
       (* only read *)
       No_effects, No_coeffects
+  | Ppoll ->
+      (* Effects to prevent poll from being optimised away *)
```

Comment:
```suggestion
      (* Explicit polling runs callbacks synchronously and lets us observe their
         effects (in particular this should not be optimised away). *)
```

Reply:
Honestly, I'd rather put this comment next to the primitive definition in `lambda.mli` and just move the `Ppoll` case together with either `Popaque` or `Pccall` here, without comment.
And I would prefer either a very terse description such as "Explicit poll" or something that properly explains what polls do, which is not restricted to running callbacks. I think the PR that introduced polls called them safepoints, which I find better at explaining what this primitive is doing.
I would propose "Introduce a safe point, where any pending actions such as GC requests and asynchronous callbacks can be handled". (The actual doc comment in `sys.mli` is nice, I think it wouldn't hurt to replicate it as documentation for the primitive itself.)

Reply:
My first draft of this change had `%safepoint`, `Psafepoint`, etc, but I decided that the name `poll` was more likely to be understood as a potential call out from Caml code to the runtime. A "safe point" might be somewhere that (for example) code could be patched, or a debugger could rely on some sort of consistent state.